### PR TITLE
fixed #11643 Custom pool properties are lost when restarted in governance mode 

### LINKED ISSUE #11643: Custom pool properties  are lost when restarted in governance mode
Hi guys,
when l use distSQL to add resource with custom pool properties , just like this 
```sql
ADD RESOURCE resource (
    URL="jdbc:mysql://127.0.0.1:3306/db3?serverTimezone=UTC&useSSL=false",
    USER=root,
    PASSWORD=root,
    PROPERTIES("maximumPoolSize"=10,"idleTimeout"="30000")
);
```

However , the custom pool properties are lost on restart;

### PR DESCRIPTION
**Pull request by lanchengx**
Fixes #11643.

Changes proposed in this pull request:
- Fixed custom pool properties are lost when restarted in governance mode

### CODE REVIEW

**Comment by tristaZero** [comment]
@lanchengx Hi welcome. @menghaoranss Could you have a review?

**Comment by menghaoranss** [comment]
> @lanchengx Hi welcome. @menghaoranss Could you have a review?

ok

File: shardingsphere-infra/shardingsphere-infra-common/src/main/java/org/apache/shardingsphere/infra/yaml/config/swapper/YamlDataSourceConfigurationSwapper.java [hunk 1]
```diff
@@ -67,6 +71,9 @@ public DataSourceConfiguration swapToDataSourceConfiguration(final Map<String, O
      */
     public Map<String, Object> swapToMap(final DataSourceConfiguration dataSourceConfig) {
         Map<String, Object> result = new HashMap<>(dataSourceConfig.getProps());
+        if (dataSourceConfig.getCustomPoolProps().isEmpty()) {
```
Comment (menghaoranss) [inline shardingsphere-infra/shardingsphere-infra-common/src/main/java/org/apache/shardingsphere/infra/yaml/config/swapper/YamlDataSourceConfigurationSwapper.java:74]: Why put when `dataSourceConfig.getCustomPoolProps()` is empty?

Comment (lanchengx) [reply shardingsphere-infra/shardingsphere-infra-common/src/main/java/org/apache/shardingsphere/infra/yaml/config/swapper/YamlDataSourceConfigurationSwapper.java:74]: Thanks , It was a mistake submitted in ffc7a212a5d3084de3b212fe7a5f011a969cd831

**Review by RaigorJiang — APPROVED** [review]
I think this PR is good enough 👍

**Comment by lanchengx** [comment]
@menghaoranss Hi , Please take the time to review it ， thanks

**Comment by lanchengx** [comment]
Hi @menghaoranss , Is there anything I missed in this PR ？

**Review by menghaoranss — APPROVED** [review]

**Pull request merged**